[PT Run > Settings plugin] Add missing cpl names and MMC/MSC commands

htcfreek opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

This PR adds
- the missing cpl command names als alternative names.
- MMC.exe as result.
- all MMC snap-ins as result.

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] **Closes:** #20423, #21041
- [x] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [x] **Tests:** Added/updated and all pass
- [x] **Localization:** All end user facing strings can be localized
- [x] **Dev docs:** Added/updated
- [ ] **New binaries:** Added on the required places
   - [ ] [JSON for signing](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ESRPSigning_core.json) for new binaries
   - [ ] [WXS for installer](https://github.com/microsoft/PowerToys/blob/main/installer/PowerToysSetup/Product.wxs) for new binaries and localization folder
   - [ ] [YML for CI pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/ci/templates/build-powertoys-steps.yml) for new test projects
   - [ ] [YML for signed pipeline](https://github.com/microsoft/PowerToys/blob/main/.pipelines/release.yml)
- [ ] **Documentation updated:** If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/windows-uwp/tree/docs/hub/powertoys) and link it here: #xxx

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments

<!-- Describe how you validated the behavior. Add automated tests wherever possible, but list manual validation steps taken as well -->
## Validation Steps Performed
Local build and Unit tests.

jaimecbernardo commented on src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/WindowsSettings.json:
```diff
@@ -1794,6 +1800,167 @@
+      "Name": "EventViewer",
+      "Type": "AppMMC",
+      "AltNames": [ "MMC_mmcexe", "MMC_eventmgmt" ],
+      "Command": "eventmgmt.msc"
```
I think this should be eventvwr.msc

jaimecbernardo commented on src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/WindowsSettings.json:
```diff
@@ -1794,6 +1800,167 @@
+      "Name": "HyperVManager",
+      "Type": "AppMMC",
+      "AltNames": [ "MMC_mmcexe", "MMC_virtmgmt", "VirtualMachine" ],
+      "Command": "virtmgmt.msc"
```
This one didn't seem to exist for me on Windows 10 or 11. Perhaps only works when the feature is turned on?
The current behavior is sort of confusing (You try to click the result and nothing happens)

htcfreek replied:
Good point.We need to add a note to the tooltip.

htcfreek replied:
Or should we remove it as it is not a built-in feature. RSAT are not added too.

jaimecbernardo replied:
Might make sense to remove it, then?

jaimecbernardo replied:
It'd be accessible through the Program plugin right?

htcfreek replied:
As Run Command it will be accessible using program plugin. If it has a start menu shortcut (can't verify) it will be available using `.Hyper-V Manager` too.

jaimecbernardo commented on src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.WindowsSettings/WindowsSettings.json:
```diff
@@ -1794,6 +1800,167 @@
+      "Name": "WmiControl",
+      "Type": "AppMMC",
+      "AltNames": [ "MMC_mmcexe", "MMC_winmgmt", "WindowsManagementInstrumentation" ],
+      "Command": "winmgmt.msc"
```
Should be WmiMgmt.msc , I think.

jaimecbernardo reviewed (changes requested):
Thanks a lot for the contribution! I think this is a welcome addition.
Gave it a try, and added some nits.

Regarding the ones which don't have an existing msc file, I think we should skip them. It just opens mmc. (Or at least have some sort of mention that it just opens msc and that the snap in will have to be added manually)

Regarding those that might not be present in the computer, perhaps show some message on failure to open it?

What do you think?

htcfreek commented:
@jaimecbernardo
Is a tool tip enough for the MMC results without file? Otherwise we have to add it to the title or implement bigger changes in the plugin code.

htcfreek commented:
@jaimecbernardo 
I did some changes:
- I fixed the two wrong msc file names.
- I removed the Hyper-V Manger. (If Hyper-V role is installed you can open it with the Program plugin as run command and as app.)
- I removed the ActiveX MMC command. (The ActiveX snap-in is broken.)
- No we show the "note" of a setting in the sub title too, if its command is `mmc.exe`.
![image](https://user-images.githubusercontent.com/61519853/218567937-fda607ef-2e52-4bfc-9c4c-7600e746b176.png)

jaimecbernardo reviewed (approved):
Changes LGTM!
Thank you for the contribution!
